pkg/pod/entrypoint.go

```diff
@@ -223,6 +239,33 @@ func init() {
 	if err != nil {
 		log.Fatalf("failed to marshal replace ready patch bytes: %v", err)
 	}
+
+	cancelAnnotationPath := "/metadata/annotations/" + strings.Replace(cancelAnnotation, "/", "~1", 1)
+	replaceCancelPatchBytes, err = json.Marshal([]jsonpatch.JsonPatchOperation{{
+		Operation: "replace",
+		Path:      cancelAnnotationPath,
+		Value:     cancelAnnotationValue,
+	}})
+	if err != nil {
+		log.Fatalf("failed to marshal replace cancel patch bytes: %v", err)
+	}
+}
+
+// CancelPod updates the Pod's annotations to signal the cancellation
+// by projecting the cancel annotation via the Downward API.
+// Do not wrap the error, return directly, because the caller needs to judge whether it is a not found error
+func CancelPod(ctx context.Context, kubeClient kubernetes.Interface, namespace, podName string) error {
+	// PATCH the Pod's annotations to replace the cancel annotation with the
+	// "CANCEL" value, to signal the pod to be cancelled.
+	_, err := kubeClient.CoreV1().Pods(namespace).Patch(ctx, podName, types.JSONPatchType, replaceCancelPatchBytes, metav1.PatchOptions{})
+	return err
+}
+
+// DeletePod delete specify pod by namespace and pod name.
+// Do not wrap the error, return directly, because the caller needs to judge whether it is a not found error
+func DeletePod(ctx context.Context, kubeClient kubernetes.Interface, namespace, podName string) error {
+	err := kubeClient.CoreV1().Pods(namespace).Delete(ctx, podName, metav1.DeleteOptions{})
```

Comment:
nit: since this is a one-line function call, is a separate function needed?

Reply:
adjusted